pkg/discover/discover.go

```diff
@@ -0,0 +1,179 @@
+package discover
+
+import (
+	"bufio"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/distributed-technologies/helm-overdrive-app-discover/pkg/logging"
+	"gopkg.in/yaml.v3"
+)
+
+// Structs for handling the information that comes from the `app.yaml`
+type App struct {
+	Argocd_app_namespace              string
+	Argocd_app_source_repo_url        string
+	Argocd_app_source_target_revision string
+	Argocd_app_source_path            string
+	Application_folder                string
+	CreateNamespace                   bool   `yaml:"createNamespace"`
+	Name                              string `yaml:"name"`
+	Namespace                         string `yaml:"namespace"`
+	Project                           string `yaml:"project"`
+	Source                            Source `yaml:"source"`
+}
+
+type Source struct {
+	Helm_repo     string `yaml:"helm_repo"`
+	Chart_name    string `yaml:"chart_name"`
+	Chart_version string `yaml:"chart_version"`
+}
+
+// Unmarshals a .yaml file into the app struct
+func (app *App) GetValuesFromYamlFile(path string) error {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
```

Comment:
The file must be closed.
```suggestion
	defer file.Close()

```

Reply:
AFAIU you only need to close the file if you use `os.open(path)`? and file is a []byte so it has not `close()` function

Reply:
Whoops